islandora_paged_content_pdf_batch.module

```diff
@@ -13,6 +13,52 @@ function islandora_paged_content_pdf_batch_form_islandora_paged_content_upload_p
   if (isset($form_state['islandora']['shared_storage'])) {
     module_load_include('inc', 'islandora', 'includes/ingest.form');
     $shared_storage = islandora_ingest_form_get_shared_storage($form_state);
-    drupal_set_message(t('Please note that large PDFs may take a long time to process. Consider using the batch ingester located in the <a href="@url">manage colection</a> interface.', array('@url' => url("islandora/object/{$shared_storage['parent']}/manage/collection"))), 'warning');
+    drupal_set_message(t('Please note that large PDFs may take a long time to process. Consider using the batch ingester located in the <a href="@url">manage colection</a> interface.', array('@url' => url("islandora/object/{$shared_storage['parent']}/manage"))), 'warning');
+  }
+}
+
+/**
+ * Implements hook_menu().
+ */
+function islandora_paged_content_pdf_batch_menu() {
+  $items = array();
+  $items['islandora/object/%islandora_object/manage/paged_content_pdf_batch'] = array(
+    'title' => 'PDF Batch',
+    'access callback' => 'islandora_paged_content_pdf_batch_menu_access',
+    'access arguments' => array(2),
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('islandora_paged_content_pdf_batch_batch_form', 2),
+    'file' => 'includes/form.inc',
+    'type' => MENU_LOCAL_ACTION,
+    'weight' => 20,
+  );
+  return $items;
+}
+
+/**
+ * Access callback for the UI upload.
+ */
+function islandora_paged_content_pdf_batch_menu_access(AbstractObject $object) {
+  if (!islandora_object_access(ISLANDORA_INGEST, $object)) {
+    return FALSE;
+  }
+  $c = 'COLLECTION_POLICY';
```

Comment:
There is very similar code as what follows elsewhere in the pull and I wonder if we could have a helper in islandora_paged_content that returns the applicable parent or false if none.